src/main/java/seedu/address/model/student/Student.java

```diff
@@ -121,6 +124,42 @@ public void removeSession(Index sessionIndex) {
         this.sessions.remove(sessionIndex.getZeroBased());
     }
 
+    /**
+     * Removes a single {@code Session} with {@code sessionDate}.
+     * Splits the remaining {@code RecurringSession} into two {@code RecurringSession}.
+     * @param sessionIndex (one-based) to be removed.
+     */
+    public void removeRecurringSession(Index sessionIndex, SessionDate sessionDate) {
```

Comment:
seems to be quite a chunky function. im unable to think of any better implementation right now, but making it less 'chunky' can be done for 1.4 right?

Reply:
Yes agreed, added to #158